rdr_service/workflow_management/nph/sms_validation.py

```diff
@@ -0,0 +1,68 @@
+from rdr_service.services.ancillary_studies.nph_incident import create_nph_incident
+
+
+class SmsValidator:
+    def __init__(self, file_path, recipient):
+        self.file_path = file_path
+        self.recipient = recipient
+
+    def validate_bmi_value(
+        self, row: dict[str, str], max_length_allowed: int = 5
+    ) -> None:
+        """If bmi str length is greater than 5 chars (including the decimal), raise value error and send slack alert."""
+        bmi_str_len = len(row.get("bmi"))
+        if bmi_str_len > max_length_allowed:
+            msg = (
+                f"Pull list Validation Error: BMI value exceeds the max length allowed of {max_length_allowed} "
+                f"and contains {bmi_str_len} characters for sample id {row.get('sample_id')}."
+                f"Check the data at: {self.file_path}."
+            )
+            create_nph_incident(slack=True, message=msg)
+            raise ValueError(
+                f"BMI str length of {bmi_str_len} exceeds the limit of {max_length_allowed} characters."
+            )
+
+    def validate_pull_list(self, data: list[dict]) -> None:
+        """Run validation checks for each row in the pull list."""
+        for row in data:
+            self.validate_bmi_value(row)
+
+    def validate_values_exist_if_biobank_id_exists(self, row) -> None:
+        """
+        For N1 generation, check if biobank id exists in a row, then all other values defined in required_columns
+        exist. If not, raise an exception and send a slack alert.
+
+        :param row: A row representing data.
+        :type row: class: 'sqlalchemy.util._collections.result'
+        :raise ValueError: If other values does not exist when biobank_id exists.
+        """
+        required_columns = [
+            row.sample_id,
+            row.sample_identifier,
+            row.sex_at_birth,
+            row.destination,
+            row.ethnicity,
+            row.race,
+            row.bmi,
+            row.diet,
+        ]
+        if row.biobank_id:
+            if not all([col for col in required_columns]):
+                msg = (
+                    f"N1 manifest for package id {row.package_id} contains biobank id "
+                    f"but is missing other value(s) for recipient {self.recipient}."
+                )
+                create_nph_incident(slack=True, message=msg)
+                raise ValueError(
+                    "Biobank ID exists in N1, but other required values are missing."
+                )
```

Comment:
What happens if you dont have a `biobank_id`? Maybe include that in the req cols and validate against all of them?

Reply:
Yep, I think including `biobank_id` in the list will have the same result. 

I think I need to add a comment in the code describing the reasoning why other values should exist if there is a biobank_id. The way I understand it is that part of the N1 manifest contains information related to each individual's sample within each compartment (well) of a plate. So if a `biobank_id` is present in a row in N1 manifest, the corresponding well box position must also contain a `sample_id` and other details of that individual. (We noticed that in some cases we had a biobank_id but not other values.) Do you think keeping it the way it is now and adding comment will make more sense here?

Reply:
Sounds good, sorry thought this was an ingestion.